scalafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala

```diff
@@ -76,7 +76,42 @@ class GitOpsImpl(private[util] val workingDirectory: AbsoluteFile)
     )
     for {
       root <- rootDir.toSeq
-      path <- exec(cmd).toOption.toSeq.flatten
+      path <- exec(cmd).getOrElse(Seq.empty)
     } yield AbsoluteFile.fromFile(new File(path), root)
   }
+
+  override def status: Seq[AbsoluteFile] = {
+    val cmd = Seq(
+      "git",
+      "status",
+      "--porcelain"
+    )
+    for {
+      root <- rootDir.toSeq
+      statusLine <- exec(cmd).getOrElse(Seq.empty)
+      file <- getFileFromGitStatusLine(statusLine).toOption.toSeq
+    } yield AbsoluteFile.fromFile(file, root)
+  }.filter(_.exists)
+
+  private final val renameStatusCode = "R"
+  private final val renameStatusArrowDelimiter = "-> "
+
+  private def extractPathPart(s: String): Try[String] =
```

Comment:
Could you add some comments that explain the behavior (and include a reference to https://git-scm.com/docs/git-status#_short_format)?
Now it's a bit complicated and worth explaining.

Reply:
Sure 👍